### PR TITLE
feat: add ErrorWithDetail

### LINKED ISSUE #225: Add Unwrap to `ErrorWithStatus` interface
The current interface is:

```go
// ErrorWithStatus is an interface that can be implemented by an error to provide
// additional information about the error.
type ErrorWithStatus interface {
	error
	StatusCode() int
}
```

However in order to supply a user defined error the user has to supply something like this:

```go
type MyError fuego.HTTPError

var _ fuego.ErrorWithStatus = MyError{}

func (e MyError) Error() string { return e.Err.Error() }

func (e MyError) StatusCode() int { return http.StatusBadRequest }

func (e MyError) Unwrap() error { return fuego.HTTPError(e) }
```

If you remove the `Unwrap` this will still compile but it will return this: 

```
400 Bad Request: 
```

Vs

```
400 Bad Request:  <the err>
```

Unwrap is requited so `errors.As` can infer the type which we want to always be `HTTPError`.  

--- 

However, this code is somewhat brittle, maybe we look at the overall design of custom errors. It feels bad to say here is the interface we need you to adhere too, but basically they have to implement Unwrap always as: `func (e MyError) Unwrap() error { return fuego.HTTPError(e) }`. I do kinda like that we want everything to be `HTTPError` as does enforce users to adhere too: RFC 9457. Maybe we just update the docs and explicitly explain that `Unwrap` must be `func (e MyError) Unwrap() error { return fuego.HTTPError(e) }` or something along those lines? 🤷 

Looking for overall thoughts.  

P.S -- I could just be completely missing how to properly do this 😅

### PR DESCRIPTION
**Pull request by dylanhitt**
Closes #225

Adds `ErrorWithDetail` interface to allow for response of detail. This is similar to the older with Info. The motivation for this is the fact that we use the detail option in the [`slog.Error`](https://github.com/go-fuego/fuego/blob/main/errors.go#L148) in the default error handler as well as in the [`HTTPError.Error()` func](https://github.com/go-fuego/fuego/blob/main/errors.go#L43). Maybe we should default into something as well in the case where the user doesn't supply it.

If this is acceptable I will update the docs now as well.

### CODE REVIEW

**Comment by dylanhitt** [comment]
@EwenQuim should I update the docs?

**Review by EwenQuim — APPROVED** [review]

**Comment by dylanhitt** [comment]
I’ll update the docs as well tomorrow.

**Comment by dylanhitt** [comment]
@EwenQuim if you don't mind looking at the doc updates that would be great. I'm not the best writer.

**Review by EwenQuim — CHANGES_REQUESTED** [review]

**Comment by EwenQuim** [comment]
> I'm not the best writer.

If you're not the best writer, then I'm a terrible one. The doc update is great!

File: examples/basic/main.go
[the remark this answers is not in the document]
Comment (dylanhitt) [reply examples/basic/main.go:1]: Done

**Comment by dylanhitt** [comment]
@EwenQuim you mind taking another look. Trying to clear my backlog

**Review by EwenQuim — APPROVED** [review]

**Comment by EwenQuim** [comment]
Nice, thanks!

Yes don't worry, Fuego is very active at the moment. But that's a good sign!

We'll be taking a break during winter holidays.

**Pull request merged**